test/com/scottyplunkett/server/HTTPResponseCodeTest.java

```diff
@@ -0,0 +1,33 @@
+package com.scottyplunkett.server;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class HTTPResponseCodeTest {
+
+    @Test
+    void retrieve() {
```

Comment:
In general, it's good to have one assertion per method.  It's not a hard rule, but it's a guideline which can reduce noise when you're debugging later.  If it produces noise in your test suite, it's useful to ask yourself to come up with creative ways through that noise, you know?

Reply:
+1

Reply:
I agree with your point, but here is my train of thought in regard to this specific case...
- The retrieve function interacts with a set of enum constants
- If I create a for each loop in the test, it would make the test function much smaller, but the same **test** line number would fail whether it were failing on retrieving 204 or 400... (although the diff, and stack trace should show which line from the actual class is failing, it would still be more work visually for a human running/debugging the test)
- If I create 12 individual test of the same function, it makes the test file longer by 44 lines (48 lines total would fit `@Test`, `void retrievesCodeXxx`, `assertEquals`, and the ending `}` for each of the 12 different assertions, adding (IMO) boilerplate that clutters what is reasonably explicit AND the coverage tool in intelliJ seems to have difficulty counting a method as covered if the test of said method does not have the same name, so that could cause disinformation in coverage reports

The way I have it test each value and the behavior in each case while maintaining readability, and without increasing verbosity, or causing potential disinformation.  I do think it couples the retrieve method to these constant values so if a failure is caused by logic in the retrieve method it might be a little unclear at first.  Definitely, open to suggestions and hearing your opinions on other ways to look at this.